sonar-core/src/main/java/org/sonar/core/platform/ComponentContainer.java

```diff
@@ -183,14 +180,14 @@ public ComponentContainer addComponent(Object component, boolean singleton) {
     return this;
   }
 
-  public ComponentContainer addExtension(@Nullable PluginMetadata plugin, Object extension) {
+  public ComponentContainer addExtension(@Nullable PluginInfo pluginInfo, Object extension) {
     Object key = componentKeys.of(extension);
     try {
       pico.as(Characteristics.CACHE).addComponent(key, extension);
     } catch (Throwable t) {
```

Comment:
:new: ![BLOCKER](https://raw.githubusercontent.com/henryju/image-hosting/master/blocker.png)Catch Exception instead of Throwable.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1181)